src/screens/SendToken/SendTokenAmount.js

```diff
@@ -207,26 +214,58 @@ class SendTokenAmount extends React.Component<Props, State> {
     });
   };
 
-  useMaxValue = () => {
-    const txFeeInWei = this.getTxFeeInWei();
+  useMaxValue = async () => {
     const { balances } = this.props;
     const { token } = this.assetData;
     const balance = getBalance(balances, token);
+    const gasLimit = await this.getGasLimit(balance); // calculate gas limit for max available balance
+    const transactionSpeed = this.getTxSpeed();
+    const txFeeInWei = this.getTxFeeInWei(transactionSpeed, gasLimit);
     const maxAmount = calculateMaxAmount(token, balance, txFeeInWei);
     this.enoughForFee = checkIfEnoughForFee(balances, txFeeInWei);
+    const amount = formatAmount(maxAmount);
     this.setState({
-      value: {
-        amount: formatAmount(maxAmount),
-      },
+      gasLimit,
+      value: { amount },
     });
   };
 
-  getTxFeeInWei = (txSpeed?: string): BigNumber => {
+  getGasLimit = (amount?: number) => {
+    // calculate either with amount in form or provided as param
+    if (!amount) {
+      ({ amount = 0 } = this._form.getValue() || {});
```

Comment:
with lodash.get it could look more readable: `amount = get(this._form.getValue(), 'amount', 0);`

Reply:
P. S. keep in mind that the form's value will be a string and the line below won't work: `if (amount === 0) return 0;`

Reply:
Fixed in newer commits 👍